packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts

```diff
@@ -101,7 +109,13 @@ export const useScaffoldEventHistory = <
   ]);
 
   return {
-    data: events?.map(addIndexedArgsToEvent),
+    data: events?.map(addIndexedArgsToEvent) as UseScaffoldEventHistoryEventData<
```

Comment:
it's not true since it can be undefined. Is it breaking something? I tried with undefined and it works fine for me.
I think you need to add `| undefined` here, or ` || []`, or add return type of `useScaffoldEventHistory` and make data optional `data?: UseScaffoldEventHistoryEventData<TContractName,
      TEventName,
      TBlockData,
      TTransactionData,
      TReceiptData
    >`

Reply:
and probably rename `UseScaffoldEventHistoryEventData` to `UseScaffoldEventHistoryData`

Reply:
Ohh yeah that name makes sense alot, also added `| undefined`   inside `UseScaffoldEventHistoryData` itself since it representing the whole `data` type. I missed that in `catch` block we are making it `undefined` 🙌

Thanks Rinat 🙌